### DISCUSSION TITLE
Dealing with anxiety when explaining code

### DISCUSSION THREAD

**Discussion by BrysonMG** [nss-day-cohort-47/C47-Discussions#72] [Discuss a topic]
Does anyone else get a lot of anxiety from talking about their code? For some reason, whether I'm talking to a single person or to a whole group, I start freaking out and my mind goes blank. Even when I fully understand what the code is doing, I struggle to talk about it to someone else, and God help me if I'm trying to talk about something I don't fully understand. Does anyone have any recommendations for how I can get over this?

**Comment by jacksonrgoodman** [comment]
I just want you to know that I feel the same way everyday 😄😃!
It's all ok, I like to think that its like a sculpture- sculptures can't be carved with one strike, but instead require regular practice, patience, and sanding.
Also... last week you helped me see a feature of Giffygram that I was having trouble with!

I think what has helped me the most is by practicing breaking every line of code down and going over it again and again. 
There's a good chance if I do that I will still forget something next time, and I will need to break it all down again.
But eventually it sticks better!

**Comment by BryanNilsen** [comment]
Absolutely... I still get anxiety, but not as much as when I was a student. I've found the more confident I am that I understand something, the less anxiety I have when walking through my code. That said, as soon as I encounter something I'm not sure exactly what's going on, the anxiety levels start to skyrocket. 

However, in my role as a Junior, I've found a lot of great ways to handle that:
- I read the lines of code almost like a book... line by line, until I get to something that leads me to another part of my code. It's almost like a Choose Your Own Adventure book... where the story jumps based on the decisions you make.
- As soon as I start to doubt myself, I'll preface what I'm about to say with something like, "I'm not 100% sure about this, but I believe it's doing... a, b, or c..."
- I remind myself not to start bullshitting my way through an explanation if I don't know and actually say, "You know what.. I'm not sure. Let's look it up" and then search for some documentation.
- Once I announce my uncertainty, the pressure and anxiety drop significantly.

At the end of the day, the more practice you have talking through your code, the more comfortable you get and have discovered ways to handle when you're not sure what's going on.

You have to crawl before you can run.
